File: Projeto 4a/atividade1/A_peter_robinson.c
// Discente Maria Eduarda Guedes Alves

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int A(int m, int n);

int main(void) {
    int i=5, m, n, R, ok=0;
    double tEmMilSeg=0, tGasto;
    clock_t t;

    for (m=0; m<i && !ok; m++) {
        for (n=0; n<i && !ok; n++) {
            t = clock();
            R = A(m, n);  
            t = clock()-t;

            tGasto = ((double)t) / CLOCKS_PER_SEC * 1000;  // Tempo gasto em milissegundos
            tEmMilSeg += tGasto;

            printf("\nTempo gasto nesta chamada: %.6lf ms", tGasto);
            printf("\nTempo acumulado: %.6lf ms", tEmMilSeg);
            printf("\nA(%d, %d): %d\n", m, n, R);

            if (tEmMilSeg >= 60000) ok=1;       // Se passar de menos de 1 minuto
            
        }
    }

    return 0;
}

int A(int m, int n) {
    if ((m>=0) && (n>=0)) {
        if      (m==0) return n+1;
        else if (m>0 && n==0) return A(m-1, 1);
        else if (m>0 && n>0)  return A(m-1, A(m, n-1));
    }
    else {
        printf("Foi informado algum valor invalido!\n\n");
        exit(-1);
    }
}

File: Projeto 4a/atividade1/A_peter_robinson.java
// Discente Maria Eduarda Guedes Alves

package atividade1;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class A_peter_robinson {
    public static void main(String[] args) {
        BufferedReader f = null;
        String linha, num[];
        int m, n;

        try {
            f = new BufferedReader(new FileReader("A_peter_robinson_file.txt"));
            
            System.err.println();
            while ((linha = f.readLine()) != null) {
                num = linha.split(" ");
                m = Integer.parseInt(num[0]);
                n = Integer.parseInt(num[1]);

                System.out.printf("A(%d, %d) = %d\n", m, n, A(m, n));
            }
            System.err.println();
        } 
        catch (IOException e) {
            System.out.println("\nERRO ao abrir o arquivo!");
            e.printStackTrace();
        } 
        finally {
            if (f != null) {
                try { f.close(); } 
                catch (IOException e) {
                    System.out.println("Erro ao fechar o arquivo!");
                }
            }
        }
    }

    private static int A(int m, int n) {
        if      (m==0)        return n+1;
        else if (m>0 && n==0) return A(m-1, 1);
        else if (m>0 && n>0)  return A(m-1, A(m, n-1));
        else {
            System.out.println("Foi informado algum valor invalido!");
            System.exit(-1);
            return -1; 
        }
    }
}
